.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,35 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  golangci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+          # setup-go v4 uses cache automatically, which conflicts with golangci-lint's cache.
+          # See https://github.com/golangci/golangci-lint-action/pull/704
+          cache: false
+      # A workspace file is needed for golangci-lint to check the sub-modules.
+      - run: make go-workspace
+      # Work around missing go:embed file which causes a typecheck error.
+      # https://github.com/golangci/golangci-lint/issues/2912
+      - run: touch test/integration/versionchecker/testdata/test_manifests.tar
+      - id: find-go-modules
```

Comment:
@wallrj do we still need this step?

Reply:
Hmm. Perhaps not...

Reply:
You're right. With the `go.work` file in place, golangci-lint seems to check sub-modules by default:
 * https://github.com/cert-manager/cert-manager/actions/runs/7126804855